aten/src/ATen/native/cuda/UpSampleBilinear2d.cu

```diff
@@ -253,9 +320,12 @@ static void upsample_bilinear2d_backward_out_cuda_template(
   Tensor grad_output = grad_output_.contiguous();
 
   grad_input.resize_({nbatch, channels, input_height, input_width});
+  // initialization to zero is required here. As we launch one thread per output
+  // element, and atomicAdd to input gradient. Given a sparse sampling case, our
+  // threads are not covering the whole input tensor.
   grad_input.zero_();
 
-  const int num_kernels = output_height * output_width;
+  const size_t num_kernels = nbatch * channels * output_height * output_width;
   const int num_threads = std::min(
```

Comment:
Have you checked perf for smaller number of threads, like 256?

Reply:
Yep, I did. Surprisingly, it doesn't to affect the perf significantly if at all.